src/richie/apps/courses/helpers.py

```diff
@@ -0,0 +1,55 @@
+"""
+Helpers that can be useful throughout Richie's courses app
+"""
+from .factories import CategoryFactory
+
+
+def create_categories(categories_info, parent, should_publish=True):
+    """
+    Create the category tree from the SUBJECTS dictionary.
+
+
+    Arguments:
+        categories_info (List): definition of the category tree to create in the following format:
+
+            [
+                {
+                    "title": "Subject",
+                    "children": [
+                        {
+                            "title": "Arts",
+                            "children": [
+                                {"title": "Architecture"},
+                                {"title": "Sculpture"},
+                            ],
+                        },
+                        {"title": "Mathematics"},
+                    ],
+                },
+                {"title": "Level"},
+            ]
+
+        page (cms.models.pagemodel.Page): Instance of a Page below which the category
+            tree is created.
+
+    Returns:
+        generator[courses.models.Category]: yield only the leaf categories of the created tree.
```

Comment:
I'm not sure we want to only be tagging leaf categories on courses. I would think we should just exclude the top-level category (meta) and allow tagging with all others.

For example because the taxonomy might not always be 100% complete for everything that exists anywhere. Otherwise we might force users to create `Misc.` children which would not be a great look on the courses that bear it.

Reply:
good question... I have the feeling that this may depend on the plateform operator... Maybe we could condition this extra filtering to a setting? WDYT @2018D?

Reply:
I propose to make the limit filter in models/course.py optional but leave the demo site as it is ie only leaf categories linked to courses.